lib/client-assets.php

```diff
@@ -306,6 +306,29 @@ function gutenberg_scripts_and_styles( $hook ) {
 	 */
 	wp_enqueue_media();
 
+	// Export REST bases for all post types.
+	$post_type_rest_base_mapping = array();
+	foreach ( get_post_types( array(), 'objects' ) as $post_type_object ) {
+		$rest_base = ! empty( $post_type_object->rest_base ) ? $post_type_object->rest_base : $post_type_object->name;
+		$post_type_rest_base_mapping[ $post_type_object->name ] = $rest_base;
+	}
+	$script = sprintf( 'wp.api.postTypeRestBaseMapping = %s;', wp_json_encode( $post_type_rest_base_mapping ) );
+	$script .= <<<JS
```

Comment:
Wondering also if we ought to split out some of the unrelated logic here, like bootstrapping `wp.api.getPostTypeModel` with all post type REST bases, or even below with getting / validating post and post type.